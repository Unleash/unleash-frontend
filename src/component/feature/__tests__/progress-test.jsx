import React from 'react';

import Progress from '../progress-component';
import renderer from 'react-test-renderer';

jest.mock('@material-ui/core');

test('renders correctly with 15% done no fallback', () => {
    const percent = 15;
    const tree = renderer.create(<Progress strokeWidth={15} percentage={percent} isFallback={false} />);

    expect(tree).toMatchSnapshot();
});

test('renders correctly with 0% done no fallback', () => {
    const percent = 0;
    const tree = renderer.create(<Progress strokeWidth={15} percentage={percent} isFallback={false} />);

    expect(tree).toMatchSnapshot();
});

test('renders correctly with 15% done with fallback', () => {
    const percent = 15;
    const tree = renderer.create(<Progress strokeWidth={15} percentage={percent} isFallback />);

    expect(tree).toMatchSnapshot();
});

test('renders correctly with 0% done with fallback', () => {
    const percent = 0;
    const tree = renderer.create(<Progress strokeWidth={15} percentage={percent} isFallback />);

    expect(tree).toMatchSnapshot();
});
