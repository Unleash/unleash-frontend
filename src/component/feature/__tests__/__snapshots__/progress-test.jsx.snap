// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with 0% done no fallback 1`] = `
<svg
  viewBox="0 0 100 100"
>
  <path
    className="trail"
    d="
      M 50,50 m 0,-42.5
      a 42.5,42.5 0 1 1 0,85
      a 42.5,42.5 0 1 1 0,-85
    "
    fillOpacity={0}
    strokeWidth={15}
  />
  <path
    className="path "
    d="
      M 50,50 m 0,-42.5
      a 42.5,42.5 0 1 1 0,85
      a 42.5,42.5 0 1 1 0,-85
    "
    fillOpacity={0}
    strokeWidth={15}
    style={
      Object {
        "strokeDasharray": "267.0353755551324px 267.0353755551324px",
        "strokeDashoffset": "267.0353755551324px",
      }
    }
  />
  <text
    className="text"
    x={50}
    y={50}
  >
    0
    %
  </text>
</svg>
`;

exports[`renders correctly with 0% done with fallback 1`] = `
<svg
  viewBox="0 0 24 24"
>
  <path
    d="M17.3,18C19,16.5 20,14.4 20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12C4,14.4 5,16.5 6.7,18C8.2,16.7 10,16 12,16C14,16 15.9,16.7 17.3,18M12,2A10,10 0 0,1 22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2M7,9A1,1 0 0,1 8,10A1,1 0 0,1 7,11A1,1 0 0,1 6,10A1,1 0 0,1 7,9M10,6A1,1 0 0,1 11,7A1,1 0 0,1 10,8A1,1 0 0,1 9,7A1,1 0 0,1 10,6M17,9A1,1 0 0,1 18,10A1,1 0 0,1 17,11A1,1 0 0,1 16,10A1,1 0 0,1 17,9M14.4,6.1C14.9,6.3 15.1,6.9 15,7.4L13.6,10.8C13.8,11.1 14,11.5 14,12A2,2 0 0,1 12,14A2,2 0 0,1 10,12C10,11 10.7,10.1 11.7,10L13.1,6.7C13.3,6.1 13.9,5.9 14.4,6.1Z"
    fill="#E0E0E0"
  />
</svg>
`;

exports[`renders correctly with 15% done no fallback 1`] = `
<svg
  viewBox="0 0 100 100"
>
  <path
    className="trail"
    d="
      M 50,50 m 0,-42.5
      a 42.5,42.5 0 1 1 0,85
      a 42.5,42.5 0 1 1 0,-85
    "
    fillOpacity={0}
    strokeWidth={15}
  />
  <path
    className="path "
    d="
      M 50,50 m 0,-42.5
      a 42.5,42.5 0 1 1 0,85
      a 42.5,42.5 0 1 1 0,-85
    "
    fillOpacity={0}
    strokeWidth={15}
    style={
      Object {
        "strokeDasharray": "267.0353755551324px 267.0353755551324px",
        "strokeDashoffset": "226.98006922186255px",
      }
    }
  />
  <text
    className="text"
    x={50}
    y={50}
  >
    15
    %
  </text>
</svg>
`;

exports[`renders correctly with 15% done with fallback 1`] = `
<svg
  viewBox="0 0 24 24"
>
  <path
    d="M17.3,18C19,16.5 20,14.4 20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12C4,14.4 5,16.5 6.7,18C8.2,16.7 10,16 12,16C14,16 15.9,16.7 17.3,18M12,2A10,10 0 0,1 22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2M7,9A1,1 0 0,1 8,10A1,1 0 0,1 7,11A1,1 0 0,1 6,10A1,1 0 0,1 7,9M10,6A1,1 0 0,1 11,7A1,1 0 0,1 10,8A1,1 0 0,1 9,7A1,1 0 0,1 10,6M17,9A1,1 0 0,1 18,10A1,1 0 0,1 17,11A1,1 0 0,1 16,10A1,1 0 0,1 17,9M14.4,6.1C14.9,6.3 15.1,6.9 15,7.4L13.6,10.8C13.8,11.1 14,11.5 14,12A2,2 0 0,1 12,14A2,2 0 0,1 10,12C10,11 10.7,10.1 11.7,10L13.1,6.7C13.3,6.1 13.9,5.9 14.4,6.1Z"
    fill="#E0E0E0"
  />
</svg>
`;
